### ISSUE TITLE
After using a YouTube and cue the next, the videos has a 24hour timer on it

### ISSUE THREAD

**Issue by sirjaydic** [christofsteel/syng#2]
I really hope im doing this right. Today I installed from FlatPak on Fedora 41 and amazing software. only used You tube to test it and after a few songs it wouldn't start next and was counting down from 24:00:00. I could close program and it would restart previous song and them move on after that was done, but would be stuck at 24 hours if the next was a you tube.

Hope this helps.

**Comment by christofsteel** [comment]
Hm, this could happen, if a YouTube video cannot be played for some reason. This can happen, if youtube changed something internally, and an older version of yt-dlp (fork of youtube-dl) is used. Since you are using the FlatPak version, updating yt-dlp is not that trivial (for all intents and purposes unsupported).
 Do you mind telling me if it hangs always at the same video? And share the link, so I can investigate?

**Comment by sirjaydic** [comment]
Was truly random. Whet to get a screen shot and have played 5 videos and no issues. Seen it pop up for a sec one time and fixed it self. So maybe, it was just a random issue. If I get more data I'll come back to let you know. Thanks for the reply..

**Comment by christofsteel** [comment]
Ah, ok I think I can guess what happened. Syng downloads the two next videos in the background. If a song is queued, but not completely downloaded, it waits for it to download.
The timer on the website is approximated, and just ticks down each second. This is only a visual representation of the internal timer. If the song is somehow delayed, it can happen, that the timer gets negative and then it starts at 24 hours. This is just a visual thing. I think I should set the timer to a minimum of 0 for the next release to avoid confusion.

If that is indeed the case, there are two ways to mitigate the delay. You can reduce the quality of the YT Videos in the advanced YT settings, or activate, that it start playback even if it is not completely buffered. 

Mind you, the latter option can lead to stuttering during playback.

**Comment by sirjaydic** [comment]
I did try the play without waiting and it wouldn't continue. I let it sit for like 10 mins while I ran outside and it just kept counting down. This session I can't get it to hang up it does show up during song change but quickly continues. 
Attached a screen shot but had to move fast cause it didn't last long. I'm gonna say I was having a bad day and can't get it to do it again. Once again. Thanks for the replied and sorry to bother you. 
![Screenshot from 2024-11-09 15-45-17](https://github.com/user-attachments/assets/cc326357-4ccf-4c32-a193-ac9fd0b58817)
